src/components/TaskItem.tsx

```diff
@@ -1,23 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 interface TaskItemProps {
   taskName: string;
+  deleteTask: () => void;
 }
 
-export default function TaskItem({ taskName }: TaskItemProps) {
+export default function TaskItem({ taskName, deleteTask }: TaskItemProps) {
+  const [inputEdit, setInputEdit] = useState<string>(taskName);
+  const [editando, setEditando] = useState<boolean>(false);
+
+  const handleEdit = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    setInputEdit(e.target.value);
+  };
+  const togleEdit = () => {
+    setEditando(!editando);
+  };
+
   return (
     <li className="todo-stack-small">
       <div className="c-cb">
         <input type="checkbox" />
-        <label className="todo-label" htmlFor="todo-0">
-          {taskName}
-        </label>
+        {editando ? (
+          <input
+            type="text"
+            value={inputEdit}
+            id="new-todo-input"
+            className="input input__lg"
+            name="new-todo-input"
+            autoComplete="off"
+            onChange={handleEdit}
+          />
+        ) : (
+          <label className="todo-label" htmlFor="todo-0">
+            {taskName}
+          </label>
+        )}
       </div>
       <div className="btn-group">
-        <button type="button" className="btn">
-          Save <span className="visually-hidden" />
+        <button type="button" className="btn" onClick={togleEdit}>
```

Comment:
> continuando comentario da linha [18](https://github.com/Guilherme-BS/todo-list/pull/3/files#diff-d3ac101ffd80b9be0b7e83e7e20cbde310abc1bad67edb9dc5d6ab1ade55066aR18)

ficando entao
```suggestion
        <button type="button" className="btn" onClick={() => setEditando(!editando)}>
```
podendo assim excluir funçao  togleEdit